lib/jobs/ipmi-job.js

```diff
@@ -199,6 +224,98 @@ function ipmiJobFactory(
+           waterline.workitems.findOne({ id: data.workItemId })
+        ])
+        .spread (function(selInfo, workObj){
+            totalEntries = parseInt(selInfo.Entries);
+            var tmpDate = selInfo["Last Add Time"].split(/\/|:| /);
+            tmpDate = tmpDate[0] +" "+ tmpDate[1] +", "+ tmpDate[2] +
+                " "+ tmpDate[3] +":"+ tmpDate[4]+":"+ tmpDate[5];
+            currentSelDate = new Date(tmpDate);
+            lastReportedSelEntryID = _.get(workObj, 
+                'config.lastReportedSelEntryID', 0);
+            lastUpdatedSelDate = _.get(workObj, 
+                'config.lastReportedSelEntryID', new Date("01 01, 0100 00:00:00"));
+
+            // reset the counter when the the SEL log has been cleared
+            if(totalEntries < lastReportedSelEntryID){
+                lastReportedSelEntryID = 0;
+            }
+            stringifiedEntries = _.range(lastReportedSelEntryID + 1, totalEntries + 1).join(' ');
+            if(currentSelDate > lastUpdatedSelDate) {
+               return self.getSelEntries(data, stringifiedEntries)
+               .tap(function() {
+                   workObj.config.lastReportedSelEntryID = totalEntries;
+                   workObj.config.lastUpdatedSelDate = currentSelDate;
+                   return workObj.save();
+               });
+            }
+            return [];
+        })
+        .then(function(unparsedSel) {
+            return parser.parseSelDataEntries(unparsedSel);
+        })
+        .then(function(parsedSel) {
+            var rawArr = [];
+            var rawData;
+            // we get the raw sel info for each sel entry in order to decode additional informaiton
+            sel = parsedSel;
+            if(sel) {
+                _.forEach(parsedSel, function(entry) {
+                    rawData = self.getRawSelEntry(data, entry["SEL Record ID"])
+                    .then(function(data){
+                        return data;
+                    });
+                    rawArr.push(rawData);
+                });
+            }
+            return Promise.all(rawArr);
+        })
+        .then(function(rawArr) {
+            // we add two properties ("Sensor Type Code" and "Event Type Code") to each sel entry.
+            var decodedSel = {
+                "Event Type Code" : "",
+                "Sensor Type Code" : ""
+            };
+            var regex = new RegExp('\r?\n','g');
+            rawArr.forEach(function (value, index) {
+                var values = [];
+                var rawSelRecord;
+                var mask = 1 << 7; // a mask to clear the 7th bit
+
+                rawArr[index] = value.replace(regex, '');
+                values = rawArr[index].split(" ");
+                rawSelRecord = values.slice(3,19);
+
+                //This is from table 32 of the IPMI spec v4
+                decodedSel["Event Type Code"] = 
+                    ("0" +(parseInt(rawSelRecord[12],16) & ~mask)).toString(16).slice(-2);
```

Comment:
@tannoa2 should probably pad it out based on the string size in case it returns 2 digits and you don't end up with `"0" + "6f" === "06f`.  

Pushed up a commit to do this.